Fix and improve help rendering

soullivaneuh opened the pull request:
In latest versions of Gitlab, image rendering on wiki is broken because not public and not handled by controllers.

Referred issues:
- #6758
- #7539
- #7610

I simplified help controller in order to handle each .md and .other files.

In a nutshell:
- Add .md file on mime_types initializers to recognize as html
- Change help_show route to be more generic (subpath in all level possible)
- New controller logic:
  - Get the requested filepath (without extension) and the format
  - If `md` format, call `render_doc` (Same logic as the old one).
  - if another format (`all`), call `send_file_data` which test if file exists and just send the content with inline attachment.

With this system, all another file (jpg, png, svg, what else ?) can be handled by help controller and image are no more broken.

I think `send_file` response can be improved (cache or another header), do you have some suggestions about it ?

Thanks.

soullivaneuh committed:
Fix and improve help rendering

jvanbaarsen commented:
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.

Soullivaneuh commented:
Hi @jvanbaarsen,

I'm sorry but I was waiting for an answer of your team.

What is missing on my PR ? I think test was resolved.

Thanks.

jvanbaarsen commented:
@Soullivaneuh Can you please make sure the HoundCI comments are fixed?

Soullivaneuh commented:
HoundCi throw only too long line.

Lines of spec test with params, or config file.

Rendering could be worst if I split them, should I split them anyway?

Thanks.

jvanbaarsen commented on spec/routing/routing_spec.rb:
```diff
@@ -73,41 +73,41 @@
-  it "to #raketasks" do
-    get("/help/raketasks/README").should route_to('help#show', category: "raketasks", file: "README")
+  it 'to #raketasks' do
+    get('/help/raketasks/README').should route_to('help#show', filepath: 'raketasks/README')
```
This line can be cut to make it shorter:
```
get('/help/raketasks/README').should 
  route_to('help#show', filepath: 'raketasks/README')

```

jvanbaarsen commented on spec/routing/routing_spec.rb:
```diff
@@ -73,41 +73,41 @@
-  it "to #markdown" do
-    get("/help/markdown/markdown").should route_to('help#show',category: "markdown", file: "markdown")
+  it 'to #markdown' do
+    get('/help/markdown/markdown').should route_to('help#show',filepath: 'markdown/markdown')
```
This line can be made shorter (also space after comma fix):
```
get('/help/markdown/markdown').should
  route_to('help#show', filepath: 'markdown/markdown')
```

jvanbaarsen commented:
@Soullivaneuh All houndCI comments can be fixed by cutting the line, I added some examples. Please cut those lines.

Soullivaneuh commented:
Work in progress.

Just done a rebase, waiting houndci to fix long lines.

Sorry for late reply.

Soullivaneuh commented:
Test failed because of GitHub ddos attack: `ssh: connect to host github.com port 22: Connection timed out`

Soullivaneuh commented:
@jvanbaarsen test should be fixed now. :+1:

Soullivaneuh commented:
Just rebase the branch to avoid conflict. Can you please review it @jvanbaarsen?

Thanks.

Soullivaneuh commented:
Just rebase the branch for merge conflicts (CHANGELOG).

Any new for the review?

Thanks.

jvanbaarsen commented:
@Soullivaneuh Looks good! Thanks!

Soullivaneuh commented:
:+1:
